File: core/recc/packet/cvt/project.py
# -*- coding: utf-8 -*-

from recc.database.struct.project import Project
from recc.packet.project import ProjectA


def project_to_answer(project: Project, group_slug: str) -> ProjectA:
    project_slug = project.slug if project.slug else ""
    return ProjectA(
        group_slug=group_slug,
        project_slug=project_slug,
        name=project.name,
        description=project.description,
        features=project.features,
        visibility=project.visibility,
        extra=project.extra,
        created_at=project.created_at,
        updated_at=project.updated_at,
    )

File: api/src/packet/project.ts
export interface ProjectA {
  group_slug: string;
  project_slug: string;
  name?: string;
  description?: string;
  features?: Array<string>;
  visibility?: number;
  extra?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
  created_at?: string;
  updated_at?: string;
}

export interface CreateProjectQ {
  group_slug: string;
  project_slug: string;
  name?: string;
  description?: string;
  features?: Array<string>;
  visibility?: number;
  extra?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
}

export interface UpdateProjectQ {
  name?: string;
  description?: string;
  features?: Array<string>;
  visibility?: number;
  extra?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
}

export interface ProjectOverviewA {
  layouts?: number;
  tables?: number;
  tasks?: number;
  members?: number;
}
